src/main/java/boot/action/ActionQueue.java

```diff
@@ -0,0 +1,59 @@
+package boot.action;
+
+import java.util.LinkedList;
+import java.util.Queue;
+
+public class ActionQueue {
+    private final Queue<EntityInsertAction<?>> insertions;
+    private final Queue<EntityUpdateAction<?>> updates;
+    private final Queue<EntityDeleteAction<?>> deletions;
+
+    public ActionQueue(Queue<EntityInsertAction<?>> insertions, Queue<EntityUpdateAction<?>> updates, Queue<EntityDeleteAction<?>> deletions) {
+        this.insertions = insertions;
+        this.updates = updates;
+        this.deletions = deletions;
+    }
+
+    public ActionQueue() {
+        this(new LinkedList<>(), new LinkedList<>(), new LinkedList<>());
+    }
+
+    public void addAction(EntityInsertAction<?> action) {
+        insertions.add(action);
+    }
```

Comment:
> 지금 테스트가 하나 깨지는데 로직 개선하면서 수정하겠습니다( Save할 때 Id 받아오는 로직을 actionQueue로 넘기면서 꼬인 것 같습니다...)

현재 테스트가 깨지는건 insert 가 flush 메서드 실행 전까지 실제로 실행되지 않다보니 id 가 없어서 생기는 문제같아요 🤔
JPA 는 id 채번 방식에 따라 전략이 나뉘는데, auto_increment 라면 insert 를 쌓지 않고 바로 실행시켜 entity 에 id 를 바인딩 시켜주는 전략을 가지고 있는데요. 해당 전략에 따른 분기부분은 현재 요구사항에는 없기 때문에 만들기에 너무 과하다 생각하시면, insert 정도만 바로바로 실행시켜주시는 방식으로 해결 해 주셔도 됩니다 😄
```suggestion
    public void addAction(EntityInsertAction<?> action) {
        insertions.add(action);
        executeAllInsertions();
    }
```

해당 코드로 바꾼 뒤 테스트를 돌려보면 한군데가 깨지는데, 해당 테스트는 한번 로직을 체크해보실 필요가 있을 것 같아요 :) 밑에 코멘트 달아 놓았습니다!

Reply:
우선 제안주신대로 변경한 뒤 문제가 발생하는 부분을 수정해보도록 하겠습니다!